lib/ansible/playbook/play.py

```diff
@@ -97,19 +98,38 @@ def __init__(self):
     def __repr__(self):
         return self.get_name()
 
+    def _validate_hosts(self, attribute, name, value):
+        # Only validate 'hosts' if a value was passed in to original data set.
+        if 'hosts' in self._ds:
+            # Check for empty data
+            if value in ([], tuple(), set(), {}, ''):
+                raise AnsibleParserError("Hosts list cannot be empty. Please check your playbook")
+
+            if is_sequence(value):
+                # Make sure each item in the sequence is a string and not None
+                for host in value:
+                    if host is None:
+                        raise AnsibleParserError("Hosts list cannot contain values of 'None'. Please check your playbook")
+                    elif not is_string(host):
```

Comment:
so this is valid
```
- hosts:
    - mailer
    - local1, local2
```
but this is not?
```
- hosts:
    - mailer
    - [local1, local2]
```

Reply:
```- hosts:
    - mailer
    - local1, [local2, localhost]
```
and then again, this is also legal

Reply:
I would also do `not host` as `''` is both a string and not None, but i would argue still an invalid host

Reply:
Mixing lists and strings with comma separated lists is messy. I'm not sure how far we want to go down the path of validating specific host entry strings.

Reply:
I don't disagree, but that is 'current behavior' and not clearly a buggy use.